app/src/main/java/com/github/onlynotesswent/model/note/Note.kt

```diff
@@ -65,4 +67,96 @@ data class Note(
       val classYear: Int,
       val publicPath: String
   )
+
+  /**
+   * Represents a list of Comments for a Note. The class is immutable. It's companion object is used
+   * to produce a new CommentCollection to get a new one.
+   */
+  class CommentCollection(val commentsList: List<Comment> = emptyList()) {
+    fun getUserComments(userId: String): List<Comment> {
+      return commentsList.filter { it.userId == userId }
+    }
+
+    companion object {
+
+      /**
+       * Generates a unique ID based on the user ID, content, and current timestamp. (There is
+       * nearly no risk of collisions as a user would have to input the exact same comment content
+       * at the exact same millisecond under a Note for both comments to have the exact same Id)
+       *
+       * @param userId The unique identifier for the user.
+       * @param content The content of the comment.
+       * @return A SHA-256 hash string that uniquely represents this comment instance.
+       */
+      private fun generateId(userId: String, content: String): String {
+        val timestamp = System.currentTimeMillis()
```

Comment:
Because we work with Firebase Firestore, I think it is better to use `Timestamp.now()` as it works with a `Timestamp` object